packages/mui-base/src/SwitchUnstyled/useSwitch.types.ts

```diff
@@ -49,3 +49,27 @@ interface UseSwitchInputSlotOwnProps {
 
 export type UseSwitchInputSlotProps<TOther = {}> = Omit<TOther, keyof UseSwitchInputSlotOwnProps> &
   UseSwitchInputSlotOwnProps;
+
+export interface UseSwitchReturnValue {
+  /**
+   * If `true`, the component is checked.
+   */
+  checked: boolean;
+  /**
+   * If `true`, the component is disabled.
+   */
+  disabled: boolean;
+  /**
+   * If `true`, the component is being focused using keyboard.
+   * @default false
+   */
+  focusVisible: boolean;
+  /**
+   * Function to get the props to place on an HTML `input` to make the switch accessible.
+   */
+  getInputProps: (otherProps?: React.HTMLAttributes<HTMLInputElement>) => UseSwitchInputSlotProps;
```

Comment:
```suggestion
  /**
   * Resolver for the input slot's props.
   * @param externalProps props for the input slot
   * @returns props that should be spread on the input slot
   */
  getInputProps: (externalProps?: React.HTMLAttributes<HTMLInputElement>) => UseSwitchInputSlotProps;
```

In order to be consistent with the other slot callbacks.